src/AuthorizeDotNet.php

```diff
@@ -0,0 +1,301 @@
+<?php
+
+/**
+ * Authorize.net payment Driver
+ *
+ * @package     Nails
+ * @subpackage  driver-invoice-authorize-net
+ * @category    Driver
+ * @author      Nails Dev Team
+ * @link
+ */
+
+namespace Nails\Invoice\Driver;
+
+use Nails\Environment;
+use Nails\Factory;
+use net\authorize\api\constants\ANetEnvironment as AuthNetConstants;
+use net\authorize\api\contract\v1 as AuthNetAPI;
+use net\authorize\api\controller as AuthNetController;
+
+class AuthorizeDotNet extends PaymentBase
+{
+    const AUTH_NET_RESPONSE_OK = 'Ok';
+
+    // --------------------------------------------------------------------------
+
+    protected $sApiMode;
+    protected $oAuthentication;
+
+    // --------------------------------------------------------------------------
+
+    /**
+     * AuthorizeDotNet constructor.
+     */
+    public function __construct()
+    {
+        parent::__construct();
+
+        $this->sApiMode = Environment::is('PRODUCTION') ? AuthNetConstants::PRODUCTION : AuthNetConstants::SANDBOX;
+
+        $this->oAuthentication = new AuthNetAPI\MerchantAuthenticationType();
+        $this->oAuthentication->setName($this->getSetting('sLoginId'));
+        $this->oAuthentication->setTransactionKey($this->getSetting('sTransactionKey'));
+    }
+
+    // --------------------------------------------------------------------------
+
+    public function getApiMode()
```

Comment:
Does this method need to be public?

Reply:
Add DocBloc

Reply:
@hellopablo I've made it public so that other classes interacting with the SDK (say an app service that needs to create customer profiles) can defer to this class for obtaining the API mode.